File: javascript/fimaccies.js
let fimacci = (x,y) => [y, x + y]

let fimaccies = (x,y,n, sink) => {
	while(n-->=0){
		[x,y] = fimacci(x,y)
		sink(y)
	}
}

fimaccies(-1,1, process.argv[2], f => console.log(f))

File: python/fimaccies.py
import sys

fimacci = lambda x,y : (y, x+y)

def fimaccies (x, y, n):
	while n>=0:
		x,y = fimacci(x,y)
		n -= 1

		print(y)

n = int(sys.argv[1])
fimaccies(-1,1,n)
